### PR TITLE
[wasm][debugger] Replace Mono.Cecil with MetadataReader.

### PR DESCRIPTION
**Pull request by thaystg**
Removing the usage of Mono.Cecil and using MetadataReader to read Assembly information and Pdb information.

### CODE REVIEW

**Review by lewing — COMMENTED** [review]
Looks really good, a couple of small suggestions.  Running tests now.

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs [hunk 1]
```diff
@@ -680,79 +677,46 @@ private async Task<bool> OnPause(SessionId sessionId, JObject args, Cancellation
...
-                            SendEvent(sessionId, "Debugger.scriptParsed", scriptSource, token);
-                        }
-                        return asm.GetMethodByToken(method_token);
+                        Log("info", $"Unable to download symbols on demand url:{downloadURL} assembly: {asm.Name}");
+                        continue;
                     }
-                    catch (Exception e)
+
+                    using Stream streamToReadFrom = await response.Content.ReadAsStreamAsync(token);
+                    asm.UpdatePdbInformation(streamToReadFrom);
```
[the remark this answers is not in the document]
Comment (thaystg) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs:709]: I was having problems with the stream being collect by the GC, that is why I copied.

[the remark this answers is not in the document]
Comment (lewing) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs:711]: ah yeah, it looks like HttpContent disposes it

**Review by lewing — APPROVED** [review]
💯

**Pull request merged**